ui/src/org/pentaho/di/ui/core/dialog/BrowserEnvironmentWarningDialog.java

```diff
@@ -0,0 +1,236 @@
+ *
+ * Created by eliofreitas on 26-01-2017.
+ */
+public class BrowserEnvironmentWarningDialog extends Dialog {
+
+  private static Class<?> PKG = BrowserEnvironmentWarningDialog.class; // for i18n purposes, needed by Translator2!!
+  private Shell shell;
+  private PropsUI props;
+  private final int margin = Const.FORM_MARGIN * 3; //15
+  private final int padding = margin * 2; //30
+
+  public BrowserEnvironmentWarningDialog( Shell parent ) {
+    super( parent, SWT.NONE );
+  }
+
+  public void showWarningDialog( EnvironmentCase environment ) {
+    int maxTextWidthUbuntu = 418;
+    int maxTextWidthWindows = 286;
+    int maxTextWidthMac = 326;
+    switch ( environment ) {
+      case UBUNTU:
+        showWarningDialog(
+            BaseMessages.getString( PKG, "BrowserEnvironmentWarningDialog.Title.Ubuntu" ),
+            BaseMessages.getString( PKG, "BrowserEnvironmentWarningDialog.Message.Ubuntu" ),
+            BaseMessages.getString( PKG, "BrowserEnvironmentWarningDialog.HelpLink.Ubuntu" ),
+            maxTextWidthUbuntu );
+        break;
+      case UBUNTU_THIN:
+        showWarningDialog(
+            BaseMessages.getString( PKG, "BrowserEnvironmentWarningDialog.Title.Ubuntu" ),
+            BaseMessages.getString( PKG, "BrowserEnvironmentWarningDialog.Message.Ubuntu.Thin" ),
+            BaseMessages.getString( PKG, "BrowserEnvironmentWarningDialog.HelpLink.Ubuntu" ),
+            maxTextWidthUbuntu );
+        break;
+      case MAC_OS_X:
+        showWarningDialog(
+            BaseMessages.getString( PKG, "BrowserEnvironmentWarningDialog.Title" ),
+            BaseMessages.getString( PKG, "BrowserEnvironmentWarningDialog.Message.Mac" ),
+            BaseMessages.getString( PKG, "BrowserEnvironmentWarningDialog.HelpLink" ),
+            maxTextWidthMac );
+        break;
+      case MAC_OS_X_THIN:
+        showWarningDialog(
+            BaseMessages.getString( PKG, "BrowserEnvironmentWarningDialog.Title" ),
+            BaseMessages.getString( PKG, "BrowserEnvironmentWarningDialog.Message.Mac.Thin" ),
+            BaseMessages.getString( PKG, "BrowserEnvironmentWarningDialog.HelpLink" ),
+            maxTextWidthMac );
+        break;
+      case WINDOWS:
+        showWarningDialog(
+            BaseMessages.getString( PKG, "BrowserEnvironmentWarningDialog.Title" ),
+            BaseMessages.getString( PKG, "BrowserEnvironmentWarningDialog.Message.Windows" ),
+            BaseMessages.getString( PKG, "BrowserEnvironmentWarningDialog.HelpLink" ),
+            maxTextWidthWindows );
+        break;
+      case WINDOWS_THIN:
+        showWarningDialog(
+            BaseMessages.getString( PKG, "BrowserEnvironmentWarningDialog.Title" ),
+            BaseMessages.getString( PKG, "BrowserEnvironmentWarningDialog.Message.Windows.Thin" ),
+            BaseMessages.getString( PKG, "BrowserEnvironmentWarningDialog.HelpLink" ),
+            maxTextWidthWindows );
+        break;
+    }
```

Comment:
probably make sense use default block for "switch" to log error or throw exception about something not expecting